pkg/apiserver/registry/networkpolicy/clustergroupmember/rest.go

```diff
@@ -71,9 +77,57 @@ func (r *REST) Get(ctx context.Context, name string, options *metav1.GetOptions)
 		memberList.EffectiveMembers = effectiveMembers
 	}
 	memberList.Name = name
-	return memberList, nil
+	memberList.TotalMembers = int64(len(memberList.EffectiveMembers))
+	err = paginateMemberList(memberList, getOptions)
+	return memberList, err
+}
+
+func (r *REST) NewGetOptions() (runtime.Object, bool, string) {
+	return &controlplane.PaginationGetOptions{}, false, ""
 }
 
 func (r *REST) NamespaceScoped() bool {
 	return false
 }
+
+// paginateMemberList returns paginated results if meaningful options are provided.
+// Paginated results are continuous only when there is no member change across multiple calls.
+// Pagination is not processed if page number or limit = 0, thus returns full member list.
+// Returns an error for invalid options; returns empty list for page number beyond the total pages range.
+func paginateMemberList(memberList *controlplane.ClusterGroupMembers, pageInfo *controlplane.PaginationGetOptions) error {
+	if memberList.TotalMembers == 0 {
+		memberList.TotalPages, memberList.CurrentPage = 0, 0
+		return nil
+	}
+	if pageInfo.Page < 0 || pageInfo.Limit < 0 {
+		return errors.NewInternalError(fmt.Errorf("received invalid page number %d or limit %d for pagination", pageInfo.Page, pageInfo.Limit))
```

Comment:
That's not the right HTTP error code
You need `NewBadRequest` here. You can also return a different error message based on which parameter is wrong.